src/libs/Parser.ts

```diff
@@ -0,0 +1,51 @@
+// eslint-disable-next-line no-restricted-imports
+import {ExpensiMark} from 'expensify-common';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '@src/ONYXKEYS';
+import Log from './Log';
+import * as ReportConnection from './ReportConnection';
+
+const accountIDToNameMap: Record<string, string> = {};
+
+Onyx.connect({
+    key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+    callback: (personalDetailsList) => {
+        Object.values(personalDetailsList ?? {}).forEach((personalDetails) => {
+            if (!personalDetails) {
+                return;
+            }
+
+            accountIDToNameMap[personalDetails.accountID] = personalDetails.login ?? String(personalDetails.accountID);
+        });
+    },
+});
+
+type Extras = {
+    reportIDToName?: Record<string, string>;
+    accountIDToName?: Record<string, string>;
+    cacheVideoAttributes?: (vidSource: string, attrs: string) => void;
+    videoAttributeCache?: Record<string, string>;
+};
+
+class ExpensiMarkWithContext extends ExpensiMark {
```

Comment:
NAB: Can we please have a comment at the top of the file, or above this class (your choice), which explains the purpose of this briefly? I think it will be helpful for future readers.